src/hotspot/share/gc/shared/memAllocator.cpp

```diff
@@ -319,15 +319,13 @@ HeapWord* MemAllocator::mem_allocate_inside_tlab_slow(Allocation& allocation) co
   if (ZeroTLAB) {
     // ..and clear it.
     Copy::zero_to_words(mem, allocation._allocated_tlab_size);
-  } else {
-    // ...and zap just allocated object.
-#ifdef ASSERT
+  } else if (ZapTLAB) {
+    // ...and zap just allocated TLAB.
```

Comment:
Rereading the patch, I also so that we have:
```
// ..and clear it.
...
// ...and zap just allocated TLAB.
```
Maybe just unify this into:
```
// ..and clear it.
...
// ...and zap it.
```
Alternatively:
```
// ..and clear just allocated TLAB.
...
// ...and zap just allocated TLAB.
```

And while you are changing this. What's up with the `..` vs `...`? (rhetoric question) I'd prefer if we just picked either of those.

This comment also applies to the duplicated code in shenandoahHeap.cpp.

Reply:
I meditated about this code a bit, and I think we have a cleaner third option, see new commit.